ambry-store/src/main/java/com/github/ambry/store/BlobStoreStats.java

```diff
@@ -521,6 +535,8 @@ private void enqueueNewValue(StoreKey key, IndexValue newValue, IndexValue origi
             indexSegment.getFile().getName(), storeId);
       }
     }
+    // The remaining index entries in keyFinalStates are DELETE tombstones left by compaction (whose associated PUT is not found)
+    updateDeleteTombstoneStats(keyFinalStates.values());
```

Comment:
I don't think we should update delete tombstone for the log segment collection here. So this function is only used for statsBasedCompaction policy to decide which log segments should be compacted. We can solely rely on the periodical collection of account stats from StatsManager's StatsAggregator thread.

Reply:
I see your point. Updating delete tombstone stats in this method doesn't bring too much overhead (the stats is already in IndexFinalState). I made this change here with hope that the delete tombstone stats is more likely to be updated.